webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -299,6 +376,152 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
+    const duplicates: Array<Duplicate> = [];
+    for (const sibling of siblings){
+      const score = this.calculateSiblingLikenessScore(sibling, props);
+      if (score !== null && score < this.duplicateThreshold){
+        duplicates.push({_id: sibling._id, name: sibling.name, reported_date: sibling.reported_date, score});
+      }
+    }
+
+    return duplicates;
+  }
+
+  // Normalize the distance by dividing by the length of the longer string. 
+  // This can make the metric more adaptable across different string lengths
+  private normalizedLevenshtein(str1: string, str2: string) {
+    const distance = this.levenshteinDistance(str1, str2);
+    const maxLen = Math.max(str1.length, str2.length);
+    return (maxLen === 0) ? 0 : (distance / maxLen);
+  }
+
+  // The Levenshtein distance is a measure of the number of edits (insertions, deletions, and substitutions) 
+  // required to change one string into another.
+  private levenshteinDistance(str1: string, str2: string) {
+    return new Levenshtein(str1, str2).distance;
+  }
+
+  // Note: Look into 'https://www.npmjs.com/package/fuzzball' to perform various matching calculations
+
+  private cleanUpDuplicateSection() {
+    const $duplicateInfoElement = $('#contact-form').find('#duplicate_info');
+    $duplicateInfoElement.empty(); // Remove all child nodes
+    $duplicateInfoElement.hide();
+  }
+
+  // Promise.allSettled is not available due to the app's javascript version
+  private allSettledFallback(promises: Promise<Exclude<any, null | undefined>>[]): Promise<ReturnType[]> {
+    return Promise.all(
+      promises.map(promise => promise
+        .then((value): SuccessType => ({ status: 'fulfilled', value }))
+        .catch((reason): FailureType => ({ status: 'rejected', reason })))
+    );
+  }
+
+  private buildDuplicateHeader(count: number){
+    return `<p>${count} potential duplicate item(s) found:</p>`;
+  }
+
+  private buildDuplicateItem(d: Duplicate, count: number){
+    const link = `#/contacts/${d._id}/edit`;
+    const date = moment(d.reported_date).format('ddd MMM DD YYYY HH:mm:ss');
+    return `
+      <p>${count}) "${d.name}" <br>
+      Created on: ${date} <br>
+      <a class="duplicate-navigate-link" href="${link}" rel="noopener">Take me there</a> 
+      </p>`;
+  }
+
+  private outputDuplicates(duplicates: Duplicate[]){
+    const $duplicateInfoElement = $('#contact-form').find('#duplicate_info');
+    $duplicateInfoElement.empty(); // Remove all child nodes
+    $duplicateInfoElement.show();
+    // TODO: create a template component where these values are fed into.
+    // TODO: Use angular instead of jQuery?
+    let content = this.buildDuplicateHeader(duplicates.length);
+    let count = 0;
+    for (const d of duplicates){
+      content += this.buildDuplicateItem(d, count+1);
+      count++;
+    }
+    $duplicateInfoElement.append(content);
+    $duplicateInfoElement.on('click', '.duplicate-navigate-link', () => {
```

Comment:
No anon functions on event listeners please. They don't get properly cleaned up by the browser. They need to be named.